src/pocketmine/Player.php

```diff
@@ -3322,131 +3301,192 @@ public function handleDataPacket(DataPacket $packet){
 						$item->setDamage(null);
 					}
 
-					if($i < 9 and $item->getId() > 0){
+					if($i < 9 and $item->getId() > 0){ //TODO: Get rid of this hack.
```

Comment:
@thebigsmileXD It's a hack because some idiot gave individual recipe ingredients counts and couldn't be bothered to fix it. Take a door for example: you assume 6 wood, yeah? Some fool made _each_ item 6 wood, so it would actually cost you 36. That's why this is here. I'm going to fix that when I rewrite crafting.